src/CSharp/CodeCracker/Usage/ValidateColorAnalyzer.cs

```diff
@@ -0,0 +1,2281 @@
+﻿using CodeCracker.CSharp.Usage.MethodAnalyzers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System;
+using System.Collections.Immutable;
+using System.Globalization;
+
+namespace CodeCracker.CSharp.Usage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ValidateColorAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Category = SupportedCategories.Usage;
+        internal const string Message = "Your htmlColor value doesn't exist.";
+        internal const string Title = "Color validation";
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId.ValidateColor.ToDiagnosticId(),
+            Title,
+            Message,
+            Category,
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true,
+            description: Description,
+            helpLinkUri: HelpLink.ForDiagnostic(DiagnosticId.ValidateColor));
+        const string Description = @"This diagnostic checks the htmlColor value and triggers if the parsing fail by throwing an exception.";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) =>
+            context.RegisterSyntaxNodeAction(Analyzer, SyntaxKind.InvocationExpression);
+
+
+        private static void Analyzer(SyntaxNodeAnalysisContext context)
+        {
+            if (context.IsGenerated()) return;
+            var invocationExpression = (InvocationExpressionSyntax)context.Node;
+
+            var memberExpresion = invocationExpression.Expression as MemberAccessExpressionSyntax;
+            if (memberExpresion?.Name?.ToString() != "FromHtml") return;
+
+            if (memberExpresion.Expression as IdentifierNameSyntax != null)
+            {
+                if (((IdentifierNameSyntax)memberExpresion.Expression).Identifier.Text != nameof(ColorTranslator)) return;
+            }
+            else
+            {
+                if (((IdentifierNameSyntax)((MemberAccessExpressionSyntax)memberExpresion.Expression).Name).Identifier.Text != nameof(ColorTranslator)) return;
+            }
```

Comment:
Instead of navigating the member access expression, it is better to as for the symbol, and be sure. There might be casts involved, etc, and the structure might not be exactly as you project in this `if else` case. See the `StringFormatAnalyzer`, it does exactly that.